# HG changeset patch
# User Ben Abbott <bpabbott@mac.com>
# Date 1224698252 14400
#      Wed Oct 22 13:57:32 2008 -0400
# Node ID bca580bbda02ba8fd6741981690964f20998e116
# Parent  22c078fd926bf99a451391e1e7b660dc75d287b0
cla.m: Fix error when no children to clear.

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,12 @@
+2008-10-22  Ben Abbott <bpabbott@mac.com>
+
+	* plot/cla.m: Fix error when no children to clear.
+
 2008-10-22  John W. Eaton  <jwe@octave.org>
 
 	* plot/allchild.m: Move call to get showhiddenhandles outside of
 	unwind_protect block.
 
 2008-10-22  David Bateman  <dbateman@free.fr>
 
 	* plot/refreshdata.m: Modify demo so that "y" is evaluated in the
diff --git a/scripts/plot/cla.m b/scripts/plot/cla.m
--- a/scripts/plot/cla.m
+++ b/scripts/plot/cla.m
@@ -59,30 +59,46 @@ function cla (varargin)
       print_usage ();
     endif
   else
     hax = gca;
     oldhax = hax;
     do_reset = false;
   end
 
-  if (do_reset)
-    hc = get (hax, "children");
-  else
-    hc = findobj (get (hax, "children"), "flat", "visible", "on");
+  hc = get (hax, "children");
+
+  if (! do_reset && ! isempty (hc))
+    hc = findobj (hc, "flat", "visible", "on");
     hc = setdiff (hc, hax);
   end
 
-  ## Delete the children of the axis.
-  delete (hc);
+  if (! isempty (hc))
+    ## Delete the children of the axis.
+    delete (hc);
+  endif
 
   ## FIXME: The defaults should be "reset()" below, but so far there is
   ## no method to determine the defaults, much less return an object's
   ## properties to their default values.  Instead make a close
   ## approximation.
 
   axes (hax);
   axis auto
 
   ## Set the current axis back to where it was upon entry.
   axes (oldhax);
 
 endfunction
+
+%!test
+%! hf = figure;
+%! unwind_protect
+%!   set (hf, "visible", "off")
+%!   clf
+%!   plot (1:10)
+%!   cla ()
+%!   kids = get (gca, "children");
+%!   cla ()
+%! unwind_protect_cleanup
+%!   close (hf)
+%! end_unwind_protect
+%! assert (numel (kids), 0)
